ProjectVoyage/Models/Chat.m

```diff
@@ -25,14 +27,38 @@ + (nonnull NSString *)parseClassName {
 + (void) postChat: (NSString * _Nullable)chatTitle withDescription:(NSString * _Nullable)chatDescription withImage:(UIImage * _Nullable )image withRecipients:(NSArray<PFUser *> *)recipients withCompletion: (PFBooleanResultBlock  _Nullable)completion {
     Chat *newChat = [Chat new];
     newChat.chatTitle = chatTitle;
+    newChat.lastOrder = -1;
     newChat.recipients = recipients;
     newChat.chatDescription = chatDescription;
-    newChat.messages = [NSMutableArray new];
     newChat.current_sender = 0;
     newChat.image = [Util getPFFileFromImage:image];
     
     //[newChat pinInBackground];
     [newChat saveInBackgroundWithBlock: completion];
 }
 
++ (NSMutableArray *) getMessagesArrayForChat:(Chat *)chat {
+    PFRelation *chatMessagesRelation = [chat relationForKey:MESSAGES];
+    PFQuery *query = [chatMessagesRelation query];
+    
+    NSArray *queryKeys = [NSArray arrayWithObjects:TEXT, SENDER, ORDER, nil];
+    [query includeKeys:queryKeys];
+    
+    [query orderByAscending:ORDER];
```

Comment:
Wondering what's your current implementation for ensuring two messages never have the same index/order